snap/snapdir/snapdir.go

```diff
@@ -53,7 +65,12 @@ func (s *SnapDir) Size() (size int64, err error) {
 	return totalSize, nil
 }
 
-func (s *SnapDir) Install(targetPath, mountDir string) (bool, error) {
+func (s *SnapDir) Install(targetPath, mountDir string, opts *snap.InstallOptions) (bool, error) {
+	// if we need to copy by policy, do that
+	if opts != nil && opts.MustCopy {
```

Comment:
this is again is too aggressive, even on uc20 system even in run mode we expect for the dir case for a symlink to be ok after seeding (it shouldn't happen during seeding), it's basically used for snap try, copying the entire thing is exactly not what we want in that case. Admittedly there are security implications if somebody uses a symlink to a different device with snap try on a grade secured system. In theory we could check if targetPath and s.path are on the same device for uc20.

Reply:
ok, I implemented something softer here